agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/service/AlleleDiseaseAnnotationService.java

```diff
@@ -60,6 +62,32 @@ public List<AlleleDiseaseAnnotation> getFiltered() {
 		
 		return ret;
 	}
-	
-	
+
+	private static boolean hasValidAlleles(AlleleDiseaseAnnotation da) {
+		List<String> alleleIds = (new AlleleRepository()).getAllAlleleIDs();
+		if (da.getInternal())
+			return false;
+		if (!alleleIds.contains(da.getSubject().getCurie()))
+			return false;
+		return true;
+	}
+
+
+	private static boolean hasValidGenes(AlleleDiseaseAnnotation da) {
+		List<String> allGeneIDs = (new GeneRepository()).getAllGeneKeys();
```

Comment:
@cmpich You can't do this. Repositories HAVE to be closed, they create persistant connection to neo that have to be closed when you are finished with them... these methods should not be static.